### PR TITLE
settings redesign: Add fts named list filter

### PR DESCRIPTION
**Pull request by sirainen**
DOV-6559

### CODE REVIEW

File: source/configuration_manual/fts/dovecot.rst [hunk 1]
```diff
@@ -232,14 +232,12 @@ Example configuration using OBOX::
     fts_dovecot = yes
   }
 
-  fts_driver = dovecot
-
-  # Fall back to built in search.
-  #fts_enforced = no
+  # Don't fall back to built in search.
```
Comment (cmouse) [inline source/configuration_manual/fts/dovecot.rst:235]: built-in?

File: source/configuration_manual/fts/dovecot.rst [hunk 2]
```diff
@@ -232,14 +232,12 @@ Example configuration using OBOX::
...
+  fts_search_read_fallback = no
 
   # Local filesystem example:
   # Use local filesystem storing FTS indexes
-  fts_dovecot {
+  fts dovecot {
     fs posix {
       prefix = %h/fts/
```
Comment (cmouse) [inline source/configuration_manual/fts/dovecot.rst:242]: doesn't this need dirs setting?

File: source/settings/plugin/fts-plugin.rst [hunk 3]
```diff
@@ -105,52 +120,36 @@ Settings
...
+.. dovecot_plugin:setting:: fts_search_add_missing
    :plugin: fts
-   :values: yes, no, body
+   :default: body-search-only
+   :values: body-search-only, yes
 
-   Require FTS indexes to perform a search? This controls what to do when
-   searching headers and what to do on error situations.
+   Should missing mails be added to FTS indexes before search? With
+   body-search-only this is done only when the search query requests searching
```
Comment (cmouse) [inline source/settings/plugin/fts-plugin.rst:133]: should backtick body-search-only

**Review by cmouse — APPROVED** [review]

File: source/configuration_manual/fts/dovecot.rst (hunk 2 above)
Comment (sirainen) [reply source/configuration_manual/fts/dovecot.rst:242]: I don't think so. I'm not sure why we use it in our CI. Maybe some tests require it. But e.g. CDMI behavior is with "dirs", while S3 behavior is without. So both should work generally.

Comment (sirainen) [reply source/configuration_manual/fts/dovecot.rst:242]: Oh, or was it that we don't want it to autodelete the parent directories.

Comment (sirainen) [reply source/configuration_manual/fts/dovecot.rst:242]: I wonder if we should just rather change the default. Probably better that way.

**Pull request merged**